src/index.ts

```diff
@@ -2494,6 +2506,200 @@ async function claimAllUnclaimedAsSenderPerChain({
+	}
+
+	if (linkDetails.tokenType !== interfaces.EPeanutLinkType.erc20) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: only erc20 tokens are currently supported for gasless deposits'
+		)
+	}
+
+	if (!linkDetails.tokenAddress || !linkDetails.tokenDecimals) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: token address and decimals must be provided'
+		)
+	}
+
+	const chain3009Info = EIP3009Tokens[linkDetails.chainId]
+	if (!chain3009Info) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: there are no known EIP-3009 compliant tokens on this chain'
+		)
+	}
+
+	const tokenDomain = chain3009Info[linkDetails.tokenAddress]
+	if (!tokenDomain) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: token with the given address does not support EIP-3009'
+		)
+	}
+
+	const peanutContract = await getContract(linkDetails.chainId.toString(), null, contractVersion)
+	const uintAmount = ethers.utils.parseUnits(linkDetails.tokenAmount.toString(), linkDetails.tokenDecimals)
+	const randomNonceInt = Math.floor(Math.random() * 1e12)
+	const randomNonceHex = '0x' + randomNonceInt.toString(16).padStart(64, '0')
+
+	const { address: pubKey20 } = generateKeysFromString(password)
+	const nonceWithPubKeyHex = ethers.utils.solidityKeccak256(['address', 'bytes32'], [pubKey20, randomNonceHex])
+
+	const nowSeconds = Math.floor(Date.now() / 1000)
+	const validAfter = BigNumber.from(nowSeconds)
+	const validBefore = validAfter.add(3600) // valid for 1 hour
+
+	const args: interfaces.IPreparedGaslessDepositArgs = {
+		chainId: linkDetails.chainId,
+		contractVersion: contractVersion,
+		tokenAddress: linkDetails.tokenAddress,
+		from: address,
+		uintAmount,
+		pubKey20,
+		nonce: randomNonceHex, // nonce without pubkey. Pubkey will be added inside the peanut contract
+		validAfter,
+		validBefore,
+	}
+
+	const message: interfaces.IPreparedEIP712Message = {
+		types: ReceiveWithAuthorizationTypes,
+		primaryType: 'ReceiveWithAuthorization',
+		domain: tokenDomain,
+		values: {
+			from: address,
+			to: peanutContract.address,
+			value: uintAmount,
+			validAfter,
+			validBefore,
+			nonce: nonceWithPubKeyHex, // nonce WITH the pubkey. This is what the user will sign
+		},
+	}
+
+	return { args, message }
+}
+
+async function executeGaslessDeposit({
+	structSigner,
+	args,
+	signature,
+}: interfaces.IExecuteGaslessDepositParams): Promise<interfaces.IExecuteGaslessDepositResponse> {
+	const contract = await getContract(String(args.chainId), structSigner.signer.provider, args.contractVersion) // get the contract instance
+	const preparedArgs: any[] = [
+		args.tokenAddress,
+		args.from,
```

Comment:
This is not in line with how our relayer works: https://github.com/GrandSchtroumpf/peanut-api/blob/main/src/relayer.ts

Also, the execute function should be very minimal, just do two things:
 1. prepare tx using prepareGaslessDeposit
 2. use signAndSubmitTx to submit it

Similar as in createLink